### PR TITLE
Feature.205

### PR DESCRIPTION
**Pull request by jicarretero**
Feature 205 -- Custom IoT Agent xml

### CODE REVIEW

File: features/steps/205.IoT_Agent_Custom.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    context.headers[headerName] = headerValue
+
+
+@step(u'I set payload as in file "{filename}"')
+def set_payload_from_file(context, filename):
+    file = join(context.data_home, filename)
+    with open(file) as f:
+        context.payload = f.read().strip('\n').replace("\n", " ")
+    print(type (context.payload))
+    pass
```
Comment (flopezag) [inline features/steps/205.IoT_Agent_Custom.py]: Why pass?

File: features/steps/205.IoT_Agent_Custom.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    except exceptions.RequestException as e:
+        raise SystemExit(e)
+
+    context.statusCode = str(response.status_code)
+    try:
+        context.response = response.json()
+    except json.decoder.JSONDecodeError:
+        context.response = response.text
+
+    pass
```
Comment (flopezag) [inline features/steps/205.IoT_Agent_Custom.py]: why pass?

File: features/steps/205.IoT_Agent_Custom.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        if context.method == "POST":
+            response = post(context.url, data=context.payload, headers=context.headers)
+        elif context.method == "PUT":
+            response = put(context.url, data=context.payload, headers=context.headers)
+        elif context.method == "PATCH":
+            response = patch(context.url, data=context.payload, headers=context.headers)
+        else:
+            raise exceptions.RequestException("Unknown method krt")
+    except exceptions.RequestException as e:
+        raise SystemExit(e)
```
Comment (flopezag) [inline features/steps/205.IoT_Agent_Custom.py]: better this one:

        raise AssertionError('A request exception occurred.')

File: features/steps/205.IoT_Agent_Custom.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        context.headers = {}
+    context.headers[headerName] = headerValue
+
+
+@step(u'I set payload as in file "{filename}"')
+def set_payload_from_file(context, filename):
+    file = join(context.data_home, filename)
+    with open(file) as f:
+        context.payload = f.read().strip('\n').replace("\n", " ")
+    print(type (context.payload))
```
Comment (flopezag) [inline features/steps/205.IoT_Agent_Custom.py]: Print will not be shown, if you want to show something use:

from sys import stdout

    stdout.write(type(context.payload))

**Review by flopezag — CHANGES_REQUESTED** [review]

File: features/steps/205.IoT_Agent_Custom.py (hunk 3 above)
Comment (jicarretero) [reply features/steps/205.IoT_Agent_Custom.py]: ok

File: features/steps/205.IoT_Agent_Custom.py (hunk 4 above)
Comment (jicarretero) [reply features/steps/205.IoT_Agent_Custom.py]: No need to print to stdout anything, so, removed

File: features/steps/205.IoT_Agent_Custom.py (hunk 1 above)
Comment (jicarretero) [reply features/steps/205.IoT_Agent_Custom.py]: ok

**Review by flopezag — APPROVED** [review]

**Pull request merged**